mlir_graphblas/mlir_builder.py

```diff
@@ -13,15 +13,38 @@
 from contextlib import contextmanager
 from collections import OrderedDict
 from .sparse_utils import MLIRSparseTensor
-from .functions import BaseFunction, _default_engine
 from .engine import parse_mlir_functions, MlirJitEngine
 from .types import Type, AliasMap
 
 from typing import Dict, List, Tuple, Sequence, Generator, Optional, Union
 
-#############
-# IR Builer #
-#############
+##############
+# IR Builder #
+##############
+
+
+class MLIRCompileError(Exception):
+    pass
+
+
+_default_engine = MlirJitEngine()
+graphblas_opt_passes = (
```

Comment:
Since `graphblas_opt_passes` is a global, it'd be nice to make it follow some convention that'll make it obvious that it's a global, e.g. having the variable name in all caps or having it start with an underscore. If we go the all caps route, perhaps we should make `_default_engine` all caps as well.